Examples/Scripts/Python/seeding.py

```diff
@@ -1,266 +1,385 @@
 #!/usr/bin/env python3
-import os
 from pathlib import Path
+from typing import Optional, Union
+from enum import Enum
+from collections import namedtuple
 
 import acts
 import acts.examples
 
 
 u = acts.UnitConstants
+SeedingAlgorithm = Enum("SeedingAlgorithm", "Default TruthSmeared TruthEstimated")
+
+TruthSeedRanges = namedtuple(
+    "TruthSeedRanges",
+    ["rho", "z", "phi", "eta", "absEta", "pt", "nHits"],
+    defaults=[(None, None)] * 7,
+)
+
+ParticleSmearingSigmas = namedtuple(
+    "ParticleSmearingSigmas",
+    ["d0", "d0PtA", "d0PtB", "z0", "z0PtA", "z0PtB", "t0", "phi", "theta", "pRel"],
+    defaults=[None] * 10,
+)
+
+SeedfinderConfigArg = namedtuple(
+    "SeedfinderConfig",
+    [
+        "maxSeedsPerSpM",
+        "cotThetaMax",
+        "sigmaScattering",
+        "radLengthPerSeed",
+        "minPt",
+        "bFieldInZ",
+        "impactMax",
+        "deltaR",  # (min,max)
+        "collisionRegion",  # (min,max)
+        "r",  # (min,max)
+        "z",  # (min,max)
+        "beamPos",  # (x,y)
+    ],
+    defaults=[None] * 7 + [(None, None)] * 5,
+)
+
+TrackParamsEstimationConfig = namedtuple(
+    "TrackParamsEstimationConfig",
+    [
+        "deltaR",  # (min,max)
+    ],
+    defaults=[(None, None)],
+)
+
+
+@acts.examples.NamedTypeArgs(
+    seedingAlgorithm=SeedingAlgorithm,
+    truthSeedRanges=TruthSeedRanges,
+    particleSmearingSigmas=ParticleSmearingSigmas,
+    seedfinderConfigArg=SeedfinderConfigArg,
+    trackParamsEstimationConfig=TrackParamsEstimationConfig,
+    logLevel=acts.logging.Level,
+)
+def addSeeding(
+    s: acts.examples.Sequencer,
+    trackingGeometry: acts.TrackingGeometry,
+    field: acts.MagneticFieldProvider,
+    geoSelectionConfigFile: Optional[Union[Path, str]] = None,
+    seedingAlgorithm: SeedingAlgorithm = SeedingAlgorithm.Default,
+    truthSeedRanges: Optional[TruthSeedRanges] = TruthSeedRanges(),
+    particleSmearingSigmas: Optional[ParticleSmearingSigmas] = ParticleSmearingSigmas(),
+    initialVarInflation: Optional[list] = None,
+    seedfinderConfigArg: Optional[SeedfinderConfigArg] = SeedfinderConfigArg(),
+    trackParamsEstimationConfig: Optional[
+        TrackParamsEstimationConfig
+    ] = TrackParamsEstimationConfig(),
```

Comment:
@paulgessinger,
These parameters all now take their defaults from the C++, so defaults are only specified in one place.
The main disadvantage of this is that they aren't always trivial to find.
I did wonder whether it would help to show the defaults in the comments (either here, or in the - currently incomplete - help text below), or if the comments getting out of sync is equally problematic.
Anyway, I suggest to deal with this in a follow-up PR.

[similarly for parameters in other routines]

Reply:
I think it's good that the defaults are only in one place. We could potentially add comments pointing to where to find these defaults.

On another note: the use of `Optional` for e.g. `truthSeedRanges` implies that you can pass `None` to these arguments. Does that actually work?

Reply:
Right. I guess what I'm saying is if you're meant to either not provide the argument at all (= use default) or provide a valid value, `Optional` isn't needed here.

Reply:
Ah, maybe I misunderstand what `Optional` means. Where should it still be used?